apis/python/src/tiledbsoma/util.py

```diff
@@ -91,3 +92,28 @@ def uri_joinpath(base: str, path: str) -> str:
                 parts[2] = parts[2] + "/" + path
 
     return urllib.parse.urlunparse(parts)
+
+
+# TODO: handle slices like `1:` or `:2`
+def ids_to_list(ids: Union[slice, List[int], pa.Array]) -> pa.ChunkedArray:
+    """
+    For the interface between ``SOMADataFrame::read`` et al. (Python) and ``SOMAReader`` (C++): the
+    ``ids`` argument to the former can be slice or list; the argument to
+    ``SOMAReader::set_dim_points`` must be a list.
+    """
+    if isinstance(ids, list):
+        return pa.chunked_array(pa.array(ids))
+    if isinstance(ids, slice):
+        if ids.start is None:
+            return pa.chunked_array(pa.array([]))
+        step = ids.step
+        if step is None:
+            if ids.start <= ids.stop:
+                step = 1
+            else:
+                step = -1
+        stop = ids.stop + step
+        return pa.chunked_array(pa.array(list(range(ids.start, stop, step))))
+    if isinstance(ids, pa.Array):
```

Comment:
The intention of supporting Arrow arrays is captured in this test (currently in a PR):
https://github.com/single-cell-data/TileDB-SOMA/blob/gspowley/obs-slice-x-test/libtiledbsoma/test/test_soma_reader.py#L162

In this test, the "ids" will be type `pa.ChunkedArray`, so it will fall through the `if isintance(...)` checks and raise an exception.